src/main/java/com/artipie/npm/proxy/HttpNpmRemote.java

```
@@ -0,0 +1,176 @@
+                        "Error occurred when process get package call: %s",
+                        throwable.getMessage()
+                    );
+                    return Maybe.empty();
+                }
+            );
+    }
+
+    @Override
+    //@checkstyle ReturnCountCheck (50 lines)
+    public Maybe<NpmAsset> loadAsset(final String path, final Path tmp) {
+        return this.vertx.fileSystem().rxOpen(
+            tmp.toAbsolutePath().toString(),
+            new OpenOptions().setSync(true).setTruncateExisting(true)
+        ).flatMapMaybe(
+            asyncfile ->
+                this.client.getAbs(
+                    String.format("%s/%s", this.config.url(), path)
+                ).as(
+                    BodyCodec.pipe(asyncfile)
+                ).rxSend().flatMapMaybe(
+                    response -> {
+                        // @checkstyle MagicNumberCheck (1 line)
+                        if (response.statusCode() == 200) {
+                            return Maybe.just(
+                                new NpmAsset(
+                                    path,
+                                    new RxFile(
+                                        tmp,
+                                        this.vertx.fileSystem()
+                                    ).flow(),
+                                    response.getHeader("Last-Modified"),
+                                    response.getHeader("Content-Type")
+                                )
+                            );
+                        } else {
+                            Logger.debug(
+                                NpmProxy.class,
+                                "Could not load asset: status code %d",
+                                response.statusCode()
+                            );
+                            return Maybe.empty();
+                        }
+                    }
+                )
+        ).onErrorResumeNext(
+            throwable -> {
```

@brastak and same here: why to hide exception?